### PR TITLE
Add CreateSnapshot handler for VolumeSnapshot support 

### PR DESCRIPTION
**Pull request by hlts2**
## WHAT

@Praveen005 have implemented CreateSnapshot handler.

This PR is mirror of https://github.com/civo/civo-csi/pull/31

**CreateSnapshot handler was previously developed as a forked repository, but the CI wasn't working due to a credential issue. Therefore, I temporarily created a new PR. I will fix the credential issue in a separate PR in the future. Sorry :pray:**

[Base code](https://github.com/civo/civo-csi/pull/38/commits/2e274ffaf8bea34b2aa69f7ee7575c6be826119e) is cherry-picked from forked [PR](https://github.com/civo/civo-csi/pull/31), so code author is not changed. 

## WHY

We plan to add support for VolumeSnapshot in the civo-csi driver.
To achieve this, we need to implement the following handlers.
  - CreateSnapshot
  - [DeleteSnapshot](https://github.com/civo/civo-csi/pull/33)
  - [ListSnapshots](https://github.com/civo/civo-csi/pull/36)

## FYI

:warning: The client-side implementation is not yet complete, so some parts are impelemented with comments. Once all the implementations are finished, I will uncomment the code.

### CODE REVIEW

**Review by rytswd — APPROVED** [review]
Minor suggestion but otherwise LGTM!

File: pkg/driver/controller_server.go [hunk 1]
```diff
@@ -586,8 +587,102 @@ func (d *Driver) ControllerGetCapabilities(context.Context, *csi.ControllerGetCa
...
+	//
+	// if len(req.GetName()) == 0 {
+	// 	return nil, status.Error(codes.InvalidArgument, "Snapshot name is required")
+	// }
+	// if len(req.GetSourceVolumeId()) == 0 {
+	// 	return nil, status.Error(codes.InvalidArgument, "SourceVolumeId is required")
+	// }
+	//
+	// snapshotName := req.GetName()
+	// sourceVolID := req.GetSourceVolumeId()
```
[the remark this answers is not in the document]
Comment (hlts2) [reply pkg/driver/controller_server.go:607]: Thank you for catching it! After I fix it, I will merge this PR! :rocket:

[the remark this answers is not in the document]
Comment (hlts2) [reply pkg/driver/controller_server.go]: I have fixed it: https://github.com/civo/civo-csi/pull/38/commits/d12f0e83463c4cd9aa32367f8e6b957c28c7d967

**Comment by hlts2** [comment]
@rytswd Thank you for your review! I will merge this PR :rocket:

**Pull request merged**